spec/controllers/fed_ax_api_controller_spec.rb

```diff
@@ -0,0 +1,240 @@
+require 'rails_helper'
+require "support/shared_controller_examples"
+
+RSpec.describe FedAxApiController, type: :controller do
+  # setting up some variables that should permit valid interactions
+  let(:valid_package) { { weight: 1.5, height: 1.5, width: 1.5, depth: 1.5, product_id: 1 } }
+  let(:another_valid_package) { { weight: 2.5, height: 2.5, width: 2.5, depth: 2.5, product_id: 2 } }
+  let(:valid_packages) { [valid_package, another_valid_package] }
+  let(:valid_origin) { { country: "US", state: "WA", city: "Seattle", zip: "98101" } }
+  let(:valid_destination) { { country: "US", state: "IL", city: "Chicago", zip: "60652", order_id: 3 } }
+  let(:shipping_choice_params) { { carrier: "UPS", service_type: "UPS Three-Day Select" } }
+
+  # setting up some variables that should break stuff
+  let(:invalid_package) { [{ weight: "apples", height: "bananas", width: "avocados", depth: "pineco" }] }
+  let(:invalid_origin) { { country: "US", state: "WA", city: "Houston", zip: "60652" } }
+  let(:invalid_destination) { { country: "US", state: "OH", city: "Boise", zip: "20002" } }
+
+  describe "GET #quote" do
+    context "valid inputs" do
+      before :each do
+        VCR.use_cassette("ShippingAPI_valid") do
+          get :quote, packages: valid_packages, origin: valid_origin, destination: valid_destination
+        end
+      end
+
+      it_behaves_like "HTTP 200 success"
+      it_behaves_like "HTTP JSON response"
+
+      context "the returned hash object" do
```

Comment:
the returned hash object isn't really a context. it is however what you are describing so perhaps a describe block would be more appropriate.